libcontainer/capabilities/capabilities.go

```diff
@@ -34,50 +35,58 @@ func init() {
 }
 
 // New creates a new Caps from the given Capabilities config. Unknown Capabilities
-// or Capabilities that are unavailable in the current environment produce an error.
+// or Capabilities that are unavailable in the current environment are ignored,
+// printing a warning instead.
 func New(capConfig *configs.Capabilities) (*Caps, error) {
 	var (
-		err error
-		c   Caps
+		err      error
+		c        Caps
+		warnings = make(map[string]struct{})
 	)
 
-	if c.caps[capability.BOUNDING], err = capSlice(capConfig.Bounding); err != nil {
-		return nil, err
-	}
-	if c.caps[capability.EFFECTIVE], err = capSlice(capConfig.Effective); err != nil {
-		return nil, err
-	}
-	if c.caps[capability.INHERITABLE], err = capSlice(capConfig.Inheritable); err != nil {
-		return nil, err
-	}
-	if c.caps[capability.PERMITTED], err = capSlice(capConfig.Permitted); err != nil {
-		return nil, err
-	}
-	if c.caps[capability.AMBIENT], err = capSlice(capConfig.Ambient); err != nil {
-		return nil, err
+	c.caps = map[capability.CapType][]capability.Cap{
+		capability.BOUNDING:    capSlice(capConfig.Bounding, warnings),
+		capability.EFFECTIVE:   capSlice(capConfig.Effective, warnings),
+		capability.INHERITABLE: capSlice(capConfig.Inheritable, warnings),
+		capability.PERMITTED:   capSlice(capConfig.Permitted, warnings),
+		capability.AMBIENT:     capSlice(capConfig.Ambient, warnings),
 	}
 	if c.pid, err = capability.NewPid2(0); err != nil {
 		return nil, err
 	}
 	if err = c.pid.Load(); err != nil {
 		return nil, err
 	}
+	printWarnings(warnings)
```

Comment:
I would just do something like
```go
logrus.Warnf("ignoring unknown or unavailable capabilities: %s", reflect.ValueOf(warnings).MapKeys())
```

for the sake of saving the LOC. Example: https://play.golang.org/p/Hl7JERnmtxY

It may be slower than your code (but maybe not) and not super readable, but in this place I guess no one cares.